src/MessagePack/Resolvers/DynamicObjectResolver.cs

```diff
@@ -136,13 +136,86 @@ static FormatterCache()
                     return;
                 }
 
-                var formatterTypeInfo = DynamicObjectTypeBuilder.BuildType(assembly, typeof(T), true); // true.
+                var formatterTypeInfo = DynamicObjectTypeBuilder.BuildType(assembly, typeof(T), true, true);
                 if (formatterTypeInfo == null) return;
 
                 formatter = (IMessagePackFormatter<T>)Activator.CreateInstance(formatterTypeInfo.AsType());
             }
         }
     }
+
+    /// <summary>
+    /// ObjectResolver by dynamic code generation, no needs MessagePackObject attribute and serialized key as int.
+    /// </summary>
+    public class DynamicContractlessIntKeyObjectResolver : IFormatterResolver
```

Comment:
ContractlessIntKey have versioning problems, so I do not include it.
At least it is outside the scope of this PR.